src/main/java/org/spongepowered/api/resource/ResourcePath.java

```diff
@@ -94,36 +95,38 @@ static ResourcePath parse(String path) throws IllegalArgumentException {
     String getPath();
 
     /**
-     * Gets the root path from this path. This is the same object as returned
-     * by the following code.
-     * <pre>
-     * List<ResourcePath> parents = getParents();
-     * if (parents.isEmpty()) {
-     *     return this;
-     * }
-     * return parents.get(parents.size() - 1);
-     * </pre>
+     * Gets the root path from this path.
      *
      * @return The root path
      */
     ResourcePath getRoot();
 
     /**
-     * Gets a list of parents for this {@link ResourcePath}. Order of parents
-     * will be closest first, with the root being the last element. If the
-     * returned list is empty, this path is the root path.
+     * Gets the immediate parent for this {@link ResourcePath}. If this path is
+     * the root path, itself is returned.
      *
-     * @return The list of parents
+     * @return The parent path
+     * @see #getRoot()
      */
-    List<ResourcePath> getParents();
+    ResourcePath getParent();
 
     /**
-     * Similar to {@link #getParents()}, but only returns the name portion of
-     * the paths.
+     * Gets a list of paths for this {@link ResourcePath}. Order of parents
+     * will be closest first, with the first element being this element and the
+     * root being the last element. If the returned list is empty, this path is
+     * the root path.
      *
-     * @return The list of names in the path.
+     * @return The list of paths
      */
-    List<String> getNames();
+    List<ResourcePath> getParts();
```

Comment:
IMO the `getNames` and `getParents` methods were good, but `getParts` doesn't seem right. A resource path cannot be split into resource path parts, because it also contains the namespace. If it would split into stings that would make sense, but then it would just be the `getNames` method.

Javadocs still says it's the parents?

Reply:
I already noticed that you fixed the javadoc.